src/DurableTask.Netherite/OrchestrationService/Partition.cs

```diff
@@ -108,26 +105,37 @@ public async Task<(long, int)> CreateOrRestoreAsync(IPartitionErrorHandler error
                 }
 
             }, useSynchronizationContext: false);
-          
+
+            // before we start the partition, we have to acquire the MaxConcurrentStarts semaphore
+            // (to prevent a host from being overwhelmed by the simultaneous start of too many partitions)
+            this.TraceHelper.TracePartitionProgress("Waiting", ref this.LastTransition, this.CurrentTimeMs, "");
             await MaxConcurrentStarts.WaitAsync();
```

Comment:
nit: can we print the capacity of `MaxConcurrentStarts` in this log somehow? If we're blocked after this log, knowing the number of competing concurrent starts would be helpful info.